src/dotnet/Azure.ClientSdk.Analyzers/Azure.ClientSdk.Analyzers/Descriptors.cs

```diff
@@ -90,6 +90,20 @@ internal class Descriptors
             "Usage",
             DiagnosticSeverity.Warning, true);
 
+        public static DiagnosticDescriptor AZC0014 = new DiagnosticDescriptor(
+            nameof(AZC0014),
+            "Avoid using banned types in public API",
+            "Types from {0} assemblies should not be exposed as part of public API surface.",
+            "Usage",
+            DiagnosticSeverity.Warning, true);
+
+        public static DiagnosticDescriptor AZC0015 = new DiagnosticDescriptor(
+            nameof(AZC0015),
+            "Unexpected client method return type.",
+            "Client methods should return Pageable<T>/AsyncPageable<T>/Operation<T>/Response/Response<T>/Task<Response>/Task<Response<T>> or other client class found {0} instead.",

```

Comment:
What about `ValueTask` and `ValueTask<T>`?

Reply:
Current guidance is to stick to Task in public APIs